test/Test/Toml/BiCode/Property.hs

```diff
@@ -0,0 +1,191 @@
+module Test.Toml.BiCode.Property where
+
+import Control.Applicative ((<|>))
+import Control.Category ((>>>))
+import Data.ByteString (ByteString)
+import Data.Hashable (Hashable)
+import Data.HashSet (HashSet)
+import Data.IntSet (IntSet)
+import Data.List.NonEmpty (NonEmpty)
+import Data.Set (Set)
+import Data.Text (Text)
+import Data.Time (Day, LocalTime, TimeOfDay, ZonedTime, zonedTimeToUTC)
+import GHC.Exts (fromList)
+import Hedgehog (Gen, forAll, tripping)
+import Numeric.Natural (Natural)
+
+import Toml (TomlBiMap, TomlCodec, (.=))
+import Toml.Bi.Code (decode, encode)
+
+import Test.Toml.Gen (PropertyTest, genBool, genDay, genDouble, genHours, genInteger, genLocal,
+                      genText, genZoned, prop)
+
+import qualified Data.ByteString.Lazy as L
+import qualified Data.Text as T
+import qualified Hedgehog.Gen as Gen
+import qualified Hedgehog.Range as Range
+import qualified Toml
+
+test_encodeDecodeProp :: PropertyTest
+test_encodeDecodeProp = prop "decode . encode == id" $ do
+    bigType <- forAll genBigType
+    tripping bigType (encode bigTypeCodec) (decode bigTypeCodec)
+
+data BigType = BigType
+    { btBool          :: Bool
+    , btInteger       :: Integer
+    , btNatural       :: Natural
+    , btInt           :: Int
+    , btWord          :: Word
+    , btDouble        :: Double
+    , btFloat         :: Float
+    , btText          :: Text
+    , btString        :: String
+    , btBS            :: ByteString
+    , btLazyBS        :: L.ByteString
```

Comment:
Lol, I just noticed that in `tomland` we have `Text`, `ByteString`, `LByteString` and `String` but not `LText`. Heh, that's awkward. Probably need to add `LText` to `tomland` as well. I've created issue for this:

* https://github.com/kowainik/tomland/issues/174